source/codegen/templates/service_helpers.mako

```diff
@@ -282,23 +296,31 @@ ${initialize_standard_input_param(function_name, parameter)}\
   parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
 %>\
 %   if common_helpers.is_enum(parameter) == True:
-%     if enums[parameter["enum"]].get("generate-mappings", False):
 <%
-  map_name = parameter["enum"].lower() + "_output_map_"
-  mapped_enum_iterator_name = parameter_name + "_omap_it"
+  has_mapped_enum = 'mapped-enum' in parameter
+  has_unmapped_enum = 'enum' in parameter
+  if has_mapped_enum:
```

Comment:
Consider: there's less duplication (and waaay less conditional logic) in the server than in the proto. But shared helpers could still make sense for input vs. output.